src/main/kotlin/no/nav/syfo/personoppgave/PersonOppgaveService.kt

```diff
@@ -77,34 +82,56 @@ class PersonOppgaveService(
         }
     }
 
-    private fun behandleLps(personoppgave: PersonOppgave, veilederIdent: String) {
-        val personFnr = personoppgave.personIdent
+    private fun behandle(personoppgave: PersonOppgave, veilederIdent: String) {
+        val behandletPersonOppgave = personoppgave.behandle(veilederIdent)
+
+        database.updatePersonOppgaveBehandlet(
+            updatedPersonoppgave = behandletPersonOppgave,
+        )
+
+        publishIfAlleOppgaverBehandlet(
+            behandletPersonOppgave = behandletPersonOppgave,
+            veilederIdent = veilederIdent,
+        )
+    }
+
+    fun publishIfAlleOppgaverBehandlet(behandletPersonOppgave: PersonOppgave, veilederIdent: String) {
+        val hasNoOtherUbehandledeOppgaverOfSameType = getUbehandledePersonOppgaver(
+            personIdent = behandletPersonOppgave.personIdent,
+            personOppgaveType = behandletPersonOppgave.type,
+        ).isEmpty()
 
-        val isOnePersonOppgaveUbehandlet = getPersonOppgaveList(personFnr)
-            .filter { it.behandletTidspunkt == null && it.type == PersonOppgaveType.OPPFOLGINGSPLANLPS }
-            .size == 1
+        val personoppgavehendelseType = behandletPersonOppgave.toHendelseType()
 
-        if (isOnePersonOppgaveUbehandlet) {
-            personoppgavehendelseProducer.sendPersonoppgavehendelse(
-                PersonoppgavehendelseType.OPPFOLGINGSPLANLPS_BISTAND_BEHANDLET,
-                personFnr,
-                personoppgave.uuid,
+        if (hasNoOtherUbehandledeOppgaverOfSameType) {
+            publishPersonoppgaveHendelse(
+                personoppgavehendelseType = personoppgavehendelseType,
+                personIdent = behandletPersonOppgave.personIdent,
+                personoppgaveUUID = behandletPersonOppgave.uuid,
             )
             LOG.info(
                 "Sent Personoppgavehendelse, {}, {}",
-                StructuredArguments.keyValue("type", PersonoppgavehendelseType.OPPFOLGINGSPLANLPS_BISTAND_BEHANDLET),
+                StructuredArguments.keyValue("type", personoppgavehendelseType),
                 StructuredArguments.keyValue("veilederident", veilederIdent)
             )
         } else {
             LOG.info(
-                "No Personoppgavehendelse sent, isOnePersonOppgaveUbehandlet=false, {}, {}",
-                StructuredArguments.keyValue("type", PersonoppgavehendelseType.OPPFOLGINGSPLANLPS_BISTAND_BEHANDLET),
+                "No Personoppgavehendelse sent, there are other oppgaver of this type that still needs behandling, {}, {}",
+                StructuredArguments.keyValue("type", personoppgavehendelseType),
                 StructuredArguments.keyValue("veilederident", veilederIdent)
             )
         }
-        database.updatePersonOppgaveBehandlet(
-            personoppgave.uuid,
-            veilederIdent
+    }
+
+    fun publishPersonoppgaveHendelse(
```

Comment:
Tanken var å ta inn `PersonOppgaveService` i de andre servicene og bruke denne funksjonen, i stedet for at hver og én av de tar inn selve produceren.